### ISSUE TITLE
Two custom metrics with filters can't be used at the same time in question

### ISSUE THREAD

**Issue by grzaks** [metabase/metabase#4371]
In one of the tables I added two custom metrics, lets call it `metricA` and `metricB`. They are both calculated from different columns and use different filters:

```
metricA: Average of fieldA, filter: fieldA >= 1
metricB: Average of fieldB, filter: fieldC value on of (X, Y, Z)
```

Now, when I make new question on the table, set some other filter (ie. date range), group by day and add `metricA` it works just fine. When I change `metricA` to `metricB` it also works fine. But if I want to use them both at the same time I'm getting an error:

> Input to compound-filter does not match schema: [nil nil (named (not (matches-some-precondition? a-clojure.lang.Cons)) "Valid filter clause")]

If I remove one of the metrics from the question it again works fine. 

Chrome 56.0 on Mac OS 10.12.2 
Postgres 9.4 database
Metabase 0.22.2 self-hosted on linux server
Metabase internal database: H2

**Comment by mazameli** [comment]
Was able to reproduce. Only happens when _all_ custom metrics are filtered in some way. The error message I saw was: `Wrong number of args (4) passed to: expand/fn--24047/filter--24052`

Server logs:
```
Feb 13 18:42:26 WARN metabase.query-processor :: Query failure: Wrong number of args (4) passed to: expand/fn--24047/filter--24052
["query_processor$assert_query_status_successful.invokeStatic(query_processor.clj:566)"
"query_processor$assert_query_status_successful.invoke(query_processor.clj:559)"
"query_processor$run_and_save_query_BANG_.invokeStatic(query_processor.clj:599)"
"query_processor$run_and_save_query_BANG_.invoke(query_processor.clj:593)"
"query_processor$dataset_query.invokeStatic(query_processor.clj:632)"
"query_processor$dataset_query.invoke(query_processor.clj:605)"
"api.dataset$fn__27136$fn__27139.invoke(dataset.clj:35)"
"api.common.internal$do_with_caught_api_exceptions.invokeStatic(internal.clj:222)"
"api.common.internal$do_with_caught_api_exceptions.invoke(internal.clj:219)"
"api.dataset$fn__27136.invokeStatic(dataset.clj:29)"
"api.dataset$fn__27136.invoke(dataset.clj:29)"
"middleware$enforce_authentication$fn__33549.invoke(middleware.clj:107)"
"api.routes$fn__33659.invokeStatic(routes.clj:46)"
"api.routes$fn__33659.invoke(routes.clj:46)"
"routes$fn__35047.invokeStatic(routes.clj:38)"
"routes$fn__35047.invoke(routes.clj:38)"
"middleware$log_api_call$fn__33639$fn__33641.invoke(middleware.clj:301)"
"middleware$log_api_call$fn__33639.invoke(middleware.clj:300)"
"middleware$add_security_headers$fn__33598.invoke(middleware.clj:234)"
"middleware$bind_current_user$fn__33553.invoke(middleware.clj:127)"]
```
No pertinent JS erros.

**Comment by camsaul** [comment]
it's probably not combining the filters together when you use more than one

**Comment by salsakran** [comment]
This is a known limitation of how we're doing multiple aggregations. We mentioned them in the "technical details" of the blog post announcing them - http://www.metabase.com/blog/Metabase-0.22

The short form is that we don't really support multiple custom metrics that don't use the same filtered rows to calculate things. We'll be supporting this in the not too distant future as part of our query builder re-design.

You can probably do this via multiple series on a dashboard card for the time being.

**Comment by grzaks** [comment]
@salsakran I believe it's another issue. After submitting the issue I did another test like this:

metricA: =Average(fieldA/60), filter: fieldC value on of (X, Y, Z)
metricB: =Average(fieldB/60), filter: fieldC value on of (X, Y, Z)

They both have the same filter, both are aggregation. According to the blog post you mentioned "divide the sum of field X by the average value of field Y" should be possible.

**Comment by salsakran** [comment]
Hmm. That sounds like an additional bug =)

What I said will apply to the situation you described originally 
```
metricA: Average of fieldA, filter: fieldA >= 1
metricB: Average of fieldB, filter: fieldC value on of (X, Y, Z)
```
even if we fix this bug...

**Comment by grzaks** [comment]
Ok, perhaps my first example wasn't perfect :)

That would be some serious limitation. I'm considering migration of our BI from Pentaho CE to Metabase and we have quite a lot of metrics with filters (because they ie. make no sense for canceled orders).

This is off-topic this issue, but we also have metrics with median and 98 percentile and we have some serious trouble implementing them in metabase.

**Comment by salsakran** [comment]
Until we complete our new query builder, you can still display multiple metrics or questions on a single card in a dashboard.

WRT medians and percentile, we don't support those in the GUI query builder, I'd suggest using sql queries for the time being.

**Comment by flamber** [comment]
I cannot reproduce on 0.35.3 - it was likely fixed in 0.26
